src/WriteLockStrategy/MariaDbMetadataLockStrategy.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * This file is part of prooph/pdo-event-store.
+ * (c) 2016-2019 prooph software GmbH <contact@prooph.de>
+ * (c) 2016-2019 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Prooph\EventStore\Pdo\WriteLockStrategy;
+
+use Prooph\EventStore\Pdo\WriteLockStrategy;
+
+final class MariaDbMetadataLockStrategy implements WriteLockStrategy
+{
+    /**
+     * @var \PDO
+     */
+    private $connection;
+
+    /**
+     * @var int
+     */
+    private $timeout;
+
+    public function __construct(\PDO $connection, int $timeout = 0xffffff)
```

Comment:
Aint that timeout so high, that it doesn't even make sense to timeout at all? It times out after 6 month.

Reply:
This is the MariaDB version of MySQL's `-1`, which MariaDB does not support. Not that it this is missing feature is documented, just not listed as a possbility on the documentation. Took me a good while to figure out why the original `-1` did not work for MariaDB

Reply:
I see, and 6 month is definitely a "-1" :)